pkgs/servers/nosql/mongodb/5.0.nix

```diff
@@ -6,22 +6,24 @@ let
   };
   variants = if stdenv.isLinux then
     {
-      version = "5.0.7";
-      sha256 = "sha256-1PeDBZJNqJXHH/cSh2e+WR0PfS/b7XuJEzkkbrRT/gc=";
+      version = "5.0.18";
+      sha256 = "sha256-tvQkDBwXYRZbIuST49JJ5T9zzYe/4BQ8ul1vUGlXHxI=";
+      patches = [ ./fix-build-with-boost-1.79-5_0-linux.patch ];
     }
-  else
+  else if stdenv.isDarwin then
     {
       version = "5.0.3"; # at least darwin has to stay on 5.0.3 until the SDK used by nixpkgs is bumped to 10.13
       sha256 = "1p9pq0dfd6lynvnz5p1c8dqp4filzrz86j840xwxwx82dm1zl6p0";
-    };
+      patches = [ ./fix-build-with-boost-1.79-5_0-darwin.patch ];
+    }
+  else {};
 in
 buildMongoDB {
   version = variants.version;
   sha256 = variants.sha256;
   patches = [
     ./forget-build-dependencies-4-4.patch
     ./asio-no-experimental-string-view-4-4.patch
-    ./fix-build-with-boost-1.79-5_0.patch
```

Comment:
```nix
  ] ++ (lib.optionals !stdenv.isLinux [
     ./fix-build-with-boost-1.79-5_0.patch
  ]) ++ lib.optionals stdenv.isLinux [
     ./fix-build-with-boost-1.79-5_0-linux.patch
  ]) ++ [
```

If we want to avoid rebuild we need to preserve same order and I think the same name (at least for darwin). I haven't tested but I think OfBorg will remove the Darwin rebuild. You will need to rename the Darwin path.

Reply:
Since linux will get a rebuild you can use the variants.patches list appended because it is more readable. And add a comit that explain why only the darwin patches is applied that way: if it need to be changed we can convert to variants.patches solution.

Reply:
I think it is better if we keep the linux/darwin differences in one place and not behind multiple conditions.